admin/src/main/resources/io/buoyant/admin/js/src/router_summary.js

```diff
@@ -115,7 +118,8 @@ define([
       var pathQuery = Query.pathQuery().allPaths().withRouter(routerName).withMetrics(["requests", "retries/total"]).build();
 
       var $retriesBarChart = $barChartEl.find(".retries-bar-chart");
-      var retriesBarChart = new BarChart($retriesBarChart, getBarChartColor);
+      var barChartHelpers = getRetryBarChartHelpers();
+      var retriesBarChart = new BarChart($retriesBarChart, barChartHelpers.getColor);
```

Comment:
OK this is great. I think we can make it even greater by tweaking how the chart is instantiated. Let's make RetriesBarChart more of a wrapper that contains the scope and less of a chart library. Here's the interface I'm thinking:
```
var retriesBarChart = new RetriesBarChart($retriesBarChart)
```
```
function RetryBarChart($container) {
  function getColor(...){...}
  function getPercent(...) {...}
  ...
  var chart = new BarChart($container, getColor);
  return {update: chart.update};
}
```